### PR TITLE
Replace isort and black with ruff

### PR DESCRIPTION
**Pull request by sandhose**
Ruff now has decent parity with black and isort, so this is going to just save us a bunch of time

This can be reviewed commit by commit.
The second commit is mainly running `ruff check --fix`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -63,6 +58,8 @@ select = [
     "W",
     # pyflakes
     "F",
+    # isort
+    "I001",
```
Comment (anoadragon453) [inline pyproject.toml:62]: Why is this needed?

**Review by anoadragon453 — APPROVED** [review]
lgtm other than a small q, thanks! :rocket:

File: pyproject.toml (hunk 1 above)
Comment (clokep) [reply pyproject.toml:62]: isort isn't enabled by default, this enables it.
